internal/controller/sriov_controller.go

```diff
@@ -0,0 +1,297 @@
+	logr := log.FromContext(ctx)
+	logr.Info("Reconciling Spectrum-X ConfigMap")
+	// Fetch the ConfigMap
+	cm := &corev1.ConfigMap{}
+	err := r.Get(ctx, req.NamespacedName, cm)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			// Return early if the object is not found.
+			return ctrl.Result{}, nil
+		}
+	}
+	// Handle deletion reconciliation loop.
+	if !cm.ObjectMeta.DeletionTimestamp.IsZero() {
+		return r.reconcileDelete(ctx, cm)
+	}
+
+	spectrumXConfig, err := config.ParseConfig(cm.Data[config.ConfigMapKey])
+	if err != nil {
+		logr.Info("Failed to parse Spectrum-X ConfigMap Data")
+		return ctrl.Result{}, err
+	}
+	logr.Info("Parsed Spectrum-X Config", "config", spectrumXConfig)
+
+	// Add finalizer if not set.
+	if !controllerutil.ContainsFinalizer(cm, spectrumXSRIOVFinalizer) {
+		logr.Info("Adding finalizer to configmap")
+		controllerutil.AddFinalizer(cm, spectrumXSRIOVFinalizer)
+		cm.ObjectMeta.ManagedFields = nil
+		cm.SetGroupVersionKind(schema.GroupVersionKind{Group: "", Version: "v1", Kind: "ConfigMap"})
+		if err := r.Client.Patch(ctx, cm, client.Apply, client.ForceOwnership, client.FieldOwner(spectrumXSRIOVControllerName)); err != nil {
+			return ctrl.Result{}, fmt.Errorf("error while adding finalizer to configmap %w", err)
+		}
+		return ctrl.Result{}, nil
+	}
+
+	err = r.reconcileSRIOVNetworkNodePolicy(ctx, spectrumXConfig)
+	if err != nil {
+		logr.Info("Failed to reconcile SRIOVNetworkNodePolicy")
+		return ctrl.Result{}, err
+	}
+
+	err = r.reconcileOVSNetwork(ctx, spectrumXConfig)
+	if err != nil {
+		logr.Info("Failed to reconcile OVSNetwork")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// reconcileOVSNetwork generates OVSNetowrks according to defined Rails
+func (r *SRIOVReconciler) reconcileOVSNetwork(ctx context.Context, cfg *config.Config) error {
+	logr := log.FromContext(ctx)
+	for _, rail := range cfg.SpectrumXNetworks.Rails {
+		network := r.generateOVSNetwork(&rail, cfg.SpectrumXNetworks.MTU)
+		logr.Info("Creating OVSNetwork", "name", network.Name)
+		if err := r.Client.Patch(ctx, network, client.Apply, client.ForceOwnership, client.FieldOwner(spectrumXSRIOVControllerName)); err != nil {
+			return fmt.Errorf("error while patching %s %s: %w", network.GetObjectKind().GroupVersionKind().String(), client.ObjectKeyFromObject(network), err)
+		}
+	}
+	return nil
+}
+
+func (r *SRIOVReconciler) getDeviceNameByRail(deviceMapping []config.RailDeviceMapping, railName string) string {
+	for _, mapping := range deviceMapping {
+		if mapping.RailName == railName {
+			return mapping.DevName
+		}
+	}
+	return ""
+}
+
+// reconcileSRIOVNetworkNodePolicy generates SRIOVNetworkNodePolicy according to defined Rails
+func (r *SRIOVReconciler) reconcileSRIOVNetworkNodePolicy(ctx context.Context, cfg *config.Config) error {
+	logr := log.FromContext(ctx)
+	for _, railDevMaping := range cfg.RailDeviceMapping {
+		devName := r.getDeviceNameByRail(cfg.RailDeviceMapping, railDevMaping.RailName)
```

Comment:
We can use directly `railDevMaping.DevName` no need to call `getDeviceNameByRail`

Reply:
done